agent/reporter_test.go

```diff
@@ -21,17 +21,25 @@ func TestReporter(t *testing.T) {
 	const ntasks = 100
 
 	var (
-		ctx      = context.Background()
-		statuses = make(map[string]*api.TaskStatus) // destination map
-		unique   = make(map[uniqueStatus]struct{})  // ensure we don't receive any status twice
-		mu       sync.Mutex
-		expected = make(map[string]*api.TaskStatus)
-		wg       sync.WaitGroup
+		// NOTE(dperny): don't increase this timeout! this test should never
```

Comment:
Is there a way to test the batching code ? Successful batch send, splitting of a batch into multiple batches, failing and retrying the failed batch ?

Reply:
this tests all of those things from the reporter side by using an `UpdateTaskStatuses` method defined on the `statusReporterFunc`. `statusReporterFunc` fails sometimes at random, which will cause the reporter to retry. This test, with no changes, already caught 1 bug where in a failed batch, older tasks were rewriting newer ones.

Reply:
Non-blocking:  would you mind adding that as a comment to the test function, or as a comment on the random function?  (e.g. it non-deterministically tests the retry logic?)

Reply:
(alternately, would it make sense to make it fail every other call to `statusReporterFunc`, so we can guarantee that the retry functionality is tested?)